### PR TITLE
Release v2.5.0

### PR DESCRIPTION
**Pull request by DavidSCN**
## How to work with this template

* [x] assign a release manager, who takes care of the process
* [x] assign each point below to a responsible person, before you continue. Use `@member`.

Only the release manager should update this post. Everybody else should comment on the PR with the progress.

## Pre-PR steps

* [x] Look over [PRs](https://github.com/precice/precice/pulls?q=is%3Apr+no%3Amilestone+is%3Aclosed) and [Issues](https://github.com/precice/precice/pulls?q=is%3Aissue+no%3Amilestone+is%3Aclosed) without an assigned version. (all)
* [x] Look over entries in [`docs/changelog`](https://github.com/precice/precice/blob/develop/docs/changelog)) (all)
   * Add missing entries, if necessary
   * Fix wording and tense
* [x] Make sure you have the latest `develop` and `main` branches locally.
* [x] Merge `main` to `develop` ( This should result in no commits )
* [x] Check code base w.r.t code formatting (run [`precice/tools/formatting/check-format`](https://github.com/precice/precice/blob/develop/tools/formatting/check-format)) and reformat if required (run [`precice/tools/formatting/format-all`](https://github.com/precice/precice/blob/develop/tools/formatting/format-all))
* [x] Create branch `release-vX.Y.Z` from develop. If needed, `git rebase develop`.
* [x] Run `tools/releasing/bumpversion.sh MAJOR.MINOR.PATCH` to bump the version
* [x] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) (all)
   * Check for merged lines
   * Add things, if necessary
   * Fix wording and tense
   * Sort the entries lexicographically
   * Extract summary
* [x] Verify the version changes in:
   * [x] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
   * [x] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)
   * [x] [debian changelog](https://github.com/precice/precice/blob/develop/tools/releasing/packaging/debian/changelog)
* [x] Commit the version bump
* [x] Push the release branch to the precice repository
* Prepare independent releases
   * [ ] [Python bindings](https://github.com/precice/python-bindings/blob/develop/docs/ReleaseGuide.md)
   * [ ] (if necessary!) [MATLAB bindings](https://github.com/precice/matlab-bindings/blob/develop/docs/ReleaseGuide.md)
   * [ ] (if necessary!) [JULIA bindings](https://github.com/precice/PreCICE.jl)

## Step by step guide

* [x] Open PR from `release-vX.Y.Z` to `main` (use [this template](https://github.com/precice/precice/blob/develop/tools/releasing/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))
* [x] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
* [x] Fix potential problems in develop (all)
* [x] Rebase the release branch on develop (version bump should be the latest commit)
* [ ] Draft message to mailing list
* [ ] Write a draft "blog post" on [Discourse](https://precice.discourse.group/)
* [ ] Update documentation (all)
   * [ ] Update [XML configuration reference](https://github.com/precice/precice.github.io/blob/master/_includes/xmlreference.md)
   * [ ] Update version in [precice/precice.github.io](https://github.com/precice/precice.github.io):
      * `_config.yml`
      * `_data/sidebars/docs_sidebar.yml`
* [ ] Approve the PR with at least two reviews (all)
* [ ] Merge PR to `main` ( use `git merge --no-ff release-vX.Y.Z` )
* [ ] Tag release on `main` `vX.Y.Z` and verify by running `git describe --tags`
* [ ] Merge back to develop and verify by running `git describe --tags`
* [ ] Tripple check that you haven't messed anything up. (You can always discard local changes)
* [ ] Push `main` and push the `vX.Y.Z` tag
* [ ] Push develop
* [ ] Wait for the release pipeline
  * [ ] [To create a new draft release on GitHub](https://github.com/precice/precice/releases)
  * [ ] To automatically generate packages for latest Debian and two latest Ubuntu LTS versions.
* [ ] Write release text
* [ ] Publish the GitHub release

## Regression Tests

Use the following branches:
* precice `release-vX.Y.Z`
* pyprecice `python-bindings-vX.Y.Z.1`
* matlab-bindings `matlab-bindings-vX.Y.Z.1`
* rest `main`

Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289 and update the table.

| State | Success | Failure | Skipped |
| --- | --- | --- | --- |
| Write | `:o:` | `:x:` | `:fast_forward:` |
| Read | :o: | :x: | :fast_forward: |

| State | Tester | Test |
| --- | --- | --- |
| :o: | @IshaanDesai | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-nutils - solid-calculix |
| :o: | @DavidSCN  | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-openfoam - solid-dealii |
| :o: | @IshaanDesai  | [perpendicular-flap](https://github.com/precice/tutorials/tree/master/perpendicular-flap) fluid-su2 - solid-fenics |
| :o: | @DavidSCN  | [multiple-perpendicular-flaps](https://github.com/precice/tutorials/tree/master/multiple-perpendicular-flaps) fluid-openfoam - solid-(left+right)-dealii |
| :o:  | @MakisH  | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-openfoam serial + parallel |
| :o: | @IshaanDesai  | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-fenics serial + parallel |
| :o: |  @uekerman   | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-nutils |
| :o: | @MakisH  | [flow-over-heated-plate-nearest-projection](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate-nearest-projection) fluid-openfoam - solid-openfoam |
| :o: | @IshaanDesai   | [flow-over-heated-plate-steady-state](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate-steady-state) fluid-openfoam - solid-codeaster |
| :o: | @MakisH  | [heat-exchanger](https://github.com/precice/tutorials/tree/master/heat-exchanger) fluid-(inner+outer)-openfoam - solid-calculix |
| :o: | @MakisH  | [partitioned-elastic-beam](https://github.com/precice/tutorials/tree/master/partitioned-elastic-beam) dirichlet-calculix - neumann-calculix |
| :o: | @uekerman  | [partitioned-heat-conduction](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction) fenics - nutils |
| :o: | @IshaanDesai  | [partitioned-heat-conduction-complex](https://github.com/precice/tutorials/tree/master/partitioned-heat-conduction-complex) fenics- fenics |
| :o: | @MakisH  | [partitioned-pipe](https://github.com/precice/tutorials/tree/master/partitioned-pipe) fluid1-openfoam-pimplefoam - fluid2-openfoam-sonicliquidfoam |
| :o: | @MakisH  | [elastic-tube-1d](https://github.com/precice/tutorials/tree/master/elastic-tube-1d) fluid-cpp - solid-python |
| :o: | @MakisH | [elastic-tube-3d](https://github.com/precice/tutorials/tree/master/elastic-tube-3d) fluid-openfoam - solid-calculix |
| :o: | @erikscheurer   | MATLAB / MATLAB [ODEs](https://github.com/precice/matlab-bindings/tree/develop/tutorial) |
| :o: | @MakisH | Solverdummy [Fortran module](https://github.com/precice/fortran-module/tree/develop/examples/solverdummy) | 
| :o: | @IshaanDesai  | Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy) |
| :o: | @erikscheurer  | Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy) |
| :o: | @erikscheurer   | Solverdummy [Julia](https://github.com/precice/PreCICE.jl/tree/develop/solverdummy) |
| :o: | @uekerman  | Alya |
|  |  @KyleDavisSA | SuperMUC |

## Post-release

* [ ] Update documentation
    * [ ] Version.h header
    * [ ] deprecation and v3
    * [ ] LCI mapping (?)
* [ ] Flag [Arch Linux AUR package](https://aur.archlinux.org/packages/precice) and dependants as out-of-date.
* [ ] Update Spack recipe
* [ ] Update pyprecice Spack
* [ ] Update Website:
    * [ ] Bump version in [`_config.yml`](https://github.com/precice/precice.github.io/blob/master/_config.yml) and [`_data/sidebars/docs_sidebar.yml`](https://github.com/precice/precice.github.io/blob/master/_data/sidebars/docs_sidebar.yml)
    * [ ] Update the [XML reference](https://github.com/precice/precice.github.io/blob/master/_includes/xmlreference.md) using `binprecice md`
    * [ ] Look over the [Roadmap](https://www.precice.org/fundamentals-roadmap.html) and update entries.

### Release new version for bindings (to ensure compatibility with newest preCICE version)

- [ ] [Fortan module](https://github.com/precice/fortran-module/compare/master...develop) @MakisH 
- [ ] [MATLAB bindings](https://github.com/precice/matlab-bindings/blob/develop/docs/ReleaseGuide.md) @IshaanDesai 
- [ ] [python bindings](https://github.com/precice/python-bindings/blob/develop/docs/ReleaseGuide.md) @IshaanDesai 
- [ ] [Julia bindings](https://github.com/precice/PreCICE.jl) @erikscheurer

### Marketing

* [ ] Finalize post on [Discourse](https://precice.discourse.group/) (all)
* [ ] Write on [Gitter](https://gitter.im/precice/Lobby) @fsimonis 
* [ ] Send discourse post to the [mailing list](https://mailman.informatik.uni-stuttgart.de/mailman/listinfo/precice) @uekerman 
* [ ] [CFD-Online](https://www.cfd-online.com/Forum/news.cgi/form/0) @MakisH 
* [ ] NADigest @KyleDavisSA 
* [ ] Post on [Twitter](https://twitter.com/preCICE_org) (additionally to the automatic) @MakisH 
* [ ] Post on [ResearchGate](https://www.researchgate.net/project/preCICE) @MakisH 
* [ ] Post in [LinkedIn Group](https://www.linkedin.com/groups/9073912/) @IshaanDesai 
* [ ] Submit a short article to the [Quartl](https://www.in.tum.de/en/i05/further-activities/quartl/) DL mid june @BenjaminRodenberg 

### Misc

* [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)

To open a new PR with this template, use this [PR template query](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md)

### CODE REVIEW

**Comment by MakisH** [comment]
The changelog does not seem to have links to the PRs this time. Would be a nice addition. Can we even automate that step?

**Comment by DavidSCN** [comment]
> The changelog does not seem to have links to the PRs this time. Would be a nice addition. Can we even automate that step?

Added, yes can be automated.

**Comment by fsimonis** [comment]
> Added, yes can be automated.

They are generated in the `bumpversion` script and are present in the Changelog.

Note that the links in the PR template point to develop, which is a bit silly, really.

**Comment by IshaanDesai** [comment]
**This comment will be continuously updated as more tests are done**

Testing using the following states of repositories:

* https://github.com/precice/tutorials/commit/a0f80bc28157a93bd6381a07a798b22805e7fe76
* https://github.com/precice/calculix-adapter/commit/3ca686fe0019c39f03dc7797e892e055dce266d4
* https://github.com/precice/openfoam-adapter/commit/14c5ca18cf9afd108ff521b89634459e2b9fde62
* https://github.com/precice/python-bindings/commit/87c1084f8321996939794018d8bc8dae958568ec
* https://github.com/precice/su2-adapter/commit/ab843878c1d43302a4f0c66e25dcb364b7787478
* https://github.com/precice/fenics-adapter/commit/8863cd9e4488da1f52f93749f320ea1a405e2a96
* 

The following cases are tested and work:

* perpendicular-flap: fluid-nutils and solid-calculix
* Python solverdummies

**Comment by DavidSCN** [comment]
> Note that the links in the PR template point to develop, which is a bit silly, really.

The [PR Template](https://github.com/precice/precice/blob/develop/tools/releasing/PULL_REQUEST_TEMPLATE/release_pull_request_template.md) is anyway not in sync with the one used the last time(s). I will update it once we are done here.

**Comment by fsimonis** [comment]
Not strictly part of the release, but I built and ran the tests of 4394bbfd20d676aec6ea562cc7875f0dc95f98bd using Spack 0.18.1 (2 weeks old).
So, no issues even with the latest dependencies on GCC 12.1.1

```
Input spec
--------------------------------
precice@2.5.0%gcc
    ^petsc~hypre

Concretized
--------------------------------
precice@2.5.0%gcc@12.1.1~ipo+mpi+petsc~python+shared build_type=RelWithDebInfo arch=linux-archrolling-zen3
    ^boost@1.79.0%gcc@12.1.1~atomic~chrono~clanglibcpp~container~context~contract~coroutine~date_time~debug~exception~fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale+log~math~mpi+multithreaded~nowide~numpy~pic+program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace+system~taggedlayout+test+thread~timer~type_erasure~versionedlayout~wave cxxstd=98 patches=a440f96 visibility=hidden arch=linux-archrolling-zen3
    ^cmake@3.23.3%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-archrolling-zen3
    ^eigen@3.4.0%gcc@12.1.1~ipo build_type=RelWithDebInfo arch=linux-archrolling-zen3
        ^cmake@3.23.3%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-archrolling-zen3
    ^libxml2@2.9.13%gcc@12.1.1~python arch=linux-archrolling-zen3
        ^libiconv@1.16%gcc@12.1.1 libs=shared,static arch=linux-archrolling-zen3
        ^pkgconf@1.8.0%gcc@12.1.1 arch=linux-archrolling-zen3
        ^xz@5.2.5%gcc@12.1.1~pic libs=shared,static arch=linux-archrolling-zen3
        ^zlib@1.2.12%gcc@12.1.1+optimize+pic+shared patches=0d38234 arch=linux-archrolling-zen3
    ^openmpi@4.1.3%gcc@12.1.1~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~pmi+romio+rsh~singularity+static+vt+wrapper-rpath fabrics=none schedulers=none arch=linux-archrolling-zen3
        ^hwloc@2.7.1%gcc@12.1.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl+pci~rocm+shared arch=linux-archrolling-zen3
            ^libpciaccess@0.16%gcc@12.1.1 arch=linux-archrolling-zen3
                ^libtool@2.4.7%gcc@12.1.1 arch=linux-archrolling-zen3
                    ^m4@1.4.19%gcc@12.1.1+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-archrolling-zen3
                        ^libsigsegv@2.13%gcc@12.1.1 arch=linux-archrolling-zen3
                ^util-macros@1.19.3%gcc@12.1.1 arch=linux-archrolling-zen3
            ^ncurses@6.2%gcc@12.1.1~symlinks+termlib abi=none arch=linux-archrolling-zen3
        ^numactl@2.0.14%gcc@12.1.1 patches=4e1d78c,62fc8a8,ff37630 arch=linux-archrolling-zen3
            ^autoconf@2.69%gcc@12.1.1 patches=35c4492,7793209,a49dd5b arch=linux-archrolling-zen3
                ^perl@5.34.1%gcc@12.1.1+cpanm+shared+threads arch=linux-archrolling-zen3
                    ^berkeley-db@18.1.40%gcc@12.1.1+cxx~docs+stl patches=b231fcc arch=linux-archrolling-zen3
                    ^bzip2@1.0.8%gcc@12.1.1~debug~pic+shared arch=linux-archrolling-zen3
                        ^diffutils@3.8%gcc@12.1.1 arch=linux-archrolling-zen3
                    ^gdbm@1.19%gcc@12.1.1 arch=linux-archrolling-zen3
                        ^readline@8.1%gcc@12.1.1 arch=linux-archrolling-zen3
            ^automake@1.16.5%gcc@12.1.1 arch=linux-archrolling-zen3
        ^openssh@9.0p1%gcc@12.1.1 arch=linux-archrolling-zen3
        ^pmix@4.1.2%gcc@12.1.1~docs+pmi_backwards_compatibility~restful arch=linux-archrolling-zen3
            ^libevent@2.1.12%gcc@12.1.1+openssl arch=linux-archrolling-zen3
                ^openssl@1.1.1o%gcc@12.1.1~docs~shared certs=system arch=linux-archrolling-zen3
    ^petsc@3.17.1%gcc@12.1.1~X~batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc~hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi~mumps~openmp~p4est~parmmg~ptscotch~random123~rocm~saws~scalapack+shared~strumpack~suite-sparse+superlu-dist~tetgen~trilinos~valgrind clanguage=C arch=linux-archrolling-zen3
        ^hdf5@1.12.2%gcc@12.1.1~cxx~fortran~hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo arch=linux-archrolling-zen3
        ^metis@5.1.0%gcc@12.1.1~gdb~int64~real64+shared build_type=Release patches=4991da9,b1225da arch=linux-archrolling-zen3
        ^openblas@0.3.20%gcc@12.1.1~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=none arch=linux-archrolling-zen3
        ^parmetis@4.0.3%gcc@12.1.1~gdb~int64~ipo+shared build_type=RelWithDebInfo patches=4f89253,50ed208,704b84f arch=linux-archrolling-zen3
        ^python@3.10.5%gcc@12.1.1+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma+nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl+tix+tkinter~ucs4+uuid+zlib patches=0d98e93,f2fd060 arch=linux-archrolling-zen3
        ^superlu-dist@7.2.0%gcc@12.1.1~cuda~int64~ipo~openmp~rocm+shared build_type=RelWithDebInfo patches=8da9e21 arch=linux-archrolling-zen3
```

**Comment by erikscheurer** [comment]
Tests with:
-    Ubuntu 20.04
-    preCICE 2.5.0 release branch: [4394bbfd2](https://github.com/precice/precice/commit/4394bbfd20d676aec6ea562cc7875f0dc95f98bd)
-    matlab bindings develop branch: [67831fd](https://github.com/precice/matlab-bindings/commit/67831fd95e7468653eb418e2b8da7af202cee3a4)
- Julia bindings 2.4.0 release branch: [7002c40](https://github.com/precice/PreCICE.jl/commit/7002c40690c2738fa74b6cd246d3a97b9a5b91dc) (There are no changes on develop)

The following cases run properly till the end and are OK:

-    Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy)
-   MATLAB / MATLAB [ODEs](https://github.com/precice/matlab-bindings/tree/develop/tutorial)
- Julia [Test suite](https://github.com/precice/PreCICE.jl/blob/develop/test/runtests.jl) (Includes solverdummy)

**Comment by uekerman** [comment]
Alya testcase works as well.

**Comment by uekerman** [comment]
Partitioned heat conduction: all combinations of `nutils` and `fenics` work.

- Python bindings: current develop: https://github.com/precice/python-bindings/commit/87c1084f8321996939794018d8bc8dae958568ec
- Tutorials: current master: https://github.com/precice/tutorials/commit/eaf55b17ae88f8b37c20ae8791467f16ed1e2f0e
- Nutils: 6.3
- FEniCS-preCICE: 1.2.0

**Comment by uekerman** [comment]
Flow over heated plate: `fluid-openfoam` + `solid-nutils` works as well

- Python bindings: current develop: https://github.com/precice/python-bindings/commit/87c1084f8321996939794018d8bc8dae958568ec
- Tutorials: current master: https://github.com/precice/tutorials/commit/eaf55b17ae88f8b37c20ae8791467f16ed1e2f0e
- Nutils: 6.3
- OpenFOAM adpater: current master: https://github.com/precice/openfoam-adapter/commit/14c5ca18cf9afd108ff521b89634459e2b9fde62

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,48 @@
...
 
+## 2.5.0
+
+- Added 3D support to Linear Cell Interpolation mapping (`<mapping:linear-cell-interpolation >/`) using tetrahedra. (https://github.com/precice/precice/pull/1337)
+- Added C and Fortran bindings for setMeshTetrahedron (`precicec_setMeshTetrahedron` and `precicef_set_tetrahedron`). (https://github.com/precice/precice/pull/1382)
+- Added a clang-tidy CI and enforced the updated clang-tidy format. (https://github.com/precice/precice/pull/999)
+- Added a new interface file `precice/Version.h`, which provides version macros `PRECICE_VERSION_MAJOR`, `PRECICE_VERSION_MINOR`, `PRECICE_VERSION_PATCH` as well as the stringified version `PRECICE_VERSION`. Use the convenience macro `PRECICE_VERSION_GREATER_EQUAL(2,4,0)` to check for compatibility. `SolverInterface.hpp` and `SolverInterfaceC.h` now include `Version.h` to simplify macro usage on older version lacking the header. (https://github.com/precice/precice/pull/1392)
+- Added checks during the establishmnet of intra participant communication in order to safeguard against common issues. (https://github.com/precice/precice/pull/1290)
```
Comment (uekerman) [inline CHANGELOG.md:11]:
```suggestion
- Added checks during the establishment of inter-participant communication in order to safeguard against common issues. (https://github.com/precice/precice/pull/1290)
```

**Review by uekerman — DISMISSED** [review]

**Review by IshaanDesai — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (DavidSCN) [reply CHANGELOG.md:11]: Ah yes, will include it in the release commit.

**Review by uekerman — DISMISSED** [review]

**Comment by fsimonis** [comment]
We still have a bunch of warnings on develop. Some consistency issues with overrides, some unused variables in tests and a bunch of signed-unsigned comparisons.

I have fixed them, question is if we want to merge this into the release or simple fix it on develop.

**Comment by DavidSCN** [comment]
> I have fixed them, question is if we want to merge this into the release or simple fix it on develop.

Considering that it is mainly syntactic sugar without any influence on the functionality, I would just put them into the release.

**Comment by MakisH** [comment]
| State | Tester | Test |
| --- | --- | --- |
| :o: | @MakisH  | [flow-over-heated-plate](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate) fluid-openfoam - solid-openfoam serial + parallel |
| :o: | @MakisH  | [flow-over-heated-plate-nearest-projection](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate-nearest-projection) fluid-openfoam - solid-openfoam |
| :o:* | @MakisH  | [heat-exchanger](https://github.com/precice/tutorials/tree/master/heat-exchanger) fluid-(inner+outer)-openfoam - solid-calculix |
| :o: | @MakisH  | [partitioned-elastic-beam](https://github.com/precice/tutorials/tree/master/partitioned-elastic-beam) dirichlet-calculix - neumann-calculix |
| :o: | @MakisH  | [partitioned-pipe](https://github.com/precice/tutorials/tree/master/partitioned-pipe) fluid1-openfoam-pimplefoam - fluid2-openfoam-sonicliquidfoam |
| :o: | @MakisH  | [elastic-tube-1d](https://github.com/precice/tutorials/tree/master/elastic-tube-1d) fluid-cpp - solid-python |
| :o:*  | @MakisH | [elastic-tube-3d](https://github.com/precice/tutorials/tree/master/elastic-tube-3d) fluid-openfoam - solid-calculix |
| :o: | @MakisH | Solverdummy [Fortran module](https://github.com/precice/fortran-module/tree/develop/examples/solverdummy) | 

Everything tested with:
- preCICE from da16bb6, based on 895a12, built in the default (Debug) mode
- OpenFOAM v2206
- OpenFOAM adapter v1.1.0
- CalculiX 2.19
- CalculiX adapter 2.19.0
- Tutorials from eaf55b (effectively v202202.0)
- Python bindings as from v2.4.0 (to check for backwards compatibility). With the latest develop (87c108), I get the same error with @uekerman on including `precice/Version.h` when building with `setup.py` and using the same local setup I was using before.

In most cases, the simulations complete and the results look reasonable.
In the case of the heat exchanger and the elastic-tube-3d, the simulations are starting and working for a bit, but will take a while to complete. I will comment again once that happens.

In particular, the heat-exchanger is already beyond `Time = 10`, meaning that https://github.com/precice/tutorials/issues/260 does not apply to me anymore.

**Comment by MakisH** [comment]
Update:
- The heat exchanger completes and everything looks reasonable.
- The elastic-tube-3d runs at least for 50 time windows (half time) and the results look reasonable.

**Comment by MakisH** [comment]
fortran-module: Update PR is ready for review: https://github.com/precice/fortran-module/pull/9

**Comment by fsimonis** [comment]
@DavidSCN #1406 is ready to be integrated in the release.

**Review by uekerman — APPROVED** [review]

**Review by fsimonis — APPROVED** [review]

**Review by MakisH — APPROVED** [review]

**Pull request merged**